### ISSUE TITLE
[Blade] Ternary fails with 'or' in string

### ISSUE THREAD

**Issue by mikerockett** [laravel/framework#17929]
- Laravel Version: 5.4.11
- PHP Version: 7.0.4

### Description:

Attempting to echo:

```
{{$donor->is_company ? 'Company, Association or Organisation' : 'Individual or Family Representative'}}
```

yields the following:

```php
echo e(isset($donor->is_company ? 'Company, Association) ? $donor->is_company ? 'Company, Association : Organisation' : 'Individual or Family Representative');
```

This only occurs when 'or' is used in a string. Removing 'or' from both strings fixes the issue.

I know this kind of information should really be used in a presenter, but this came about in initial blueprint testing, which is where it was used.

I am aware that the `or` operator will be removed soon, but is there a chance this can be fixed in 5.4 anyway?

**Comment by alexsoft** [comment]
@mikerockett I think the better way to echo what you need is
```
@if ($donor->is_company)
    Company, Association or Organisation
@else
    Individual or Family Representative
@endif
```

**Comment by mikerockett** [comment]
@alexsoft The _better_ way is to use a model presenter, especially in this case. However, using a ternary should work as expected, but the blade compile is picking this up incorrectly.

**Comment by GrahamCampbell** [comment]
This is a known bug I'm afraid, and will eventually be fixed with the removal of this syntax from blade.

**Comment by GrahamCampbell** [comment]
> I am aware that the or operator will be removed soon, but is there a chance this can be fixed in 5.4 anyway?

Unlikely to be actually removed from L5.5, just from the docs for now.

**Comment by mikerockett** [comment]
@GrahamCampbell - Thanks for the clarification. Sorry if I have actually brought up an old issue, did look in the closed issues for a bit.

Perhaps the docs should be updated to show that an issue does indeed exist?
